### PR TITLE
[5.8] [WIP] Classname validator

### PR DESCRIPTION
**Pull request by liam-wiltshire**
As discussed in #28942 , this now has a `validClassName` method in the Str class, and checks on all generator commands and the migration make command.

Needs tests adding for the generator commands, but would appreciate someone eyeballing it before I spend time creating a test for each command!

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Thanks for the PR, however this has already been proposed before and implemented multiple times, and rejected. Please search for previous PRs, and you will find them. :)

**Comment by GrahamCampbell** [comment]
Maybe this time will be different though. ;)

**Comment by liam-wiltshire** [comment]
@GrahamCampbell , this came off the back of my previous PR where @taylorotwell indicated he _might_ consider it if it covered all the classes. See: https://github.com/laravel/framework/pull/28942#issuecomment-505449742

That said, this is why I wanted someone to eyeball it before I spend time putting the extra tests together - not much point if it gets closed.

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -590,6 +590,33 @@ public static function orderedUuid()
...
+     *
+     * @see https://www.php.net/manual/en/language.oop5.basic.php
+     *
+     * @param string $className
+     * @return bool
+     */
+    public static function validClassName($className)
+    {
+        // This may be a FQCN, so we split it into parts to check each
+        $parts = explode('\\', ltrim($className, '\\'));
```
Comment (timacdonald) [inline src/Illuminate/Support/Str.php]: the `ltrim($className, '\\')` should probably only remove the first backslash.

i.e. Str::validClassName("\\\\\\\\\test");

Comment (liam-wiltshire) [reply src/Illuminate/Support/Str.php]: That is a very good point - thanks :-)

**Comment by taylorotwell** [comment]
Hmm, I'll just hold off for now I think. Thanks for the idea though. 👍

**Pull request closed without merging**